pkgs/os-specific/linux/esdm/default.nix

```diff
@@ -0,0 +1,83 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, protobufc
+, pkg-config
+, fuse3
+, meson
+, ninja
+, libselinux
+, jitterentropy
+  # A more detailed explaination of the following meson build options can be found
+  # in the source code of esdm.
+  # A brief explanation is given:
+, selinux ? false # enable selinux support
+, drng_hash_drbg ? true  # set the default drng callback
+, drng_chacha20 ? false # set the default drng callback
+, ais2031 ? true # set the seeding strategy to be compliant with AIS 20/31
+, linux-devfiles ? true # enable linux /dev/random and /dev/urandom support
+, linux-getrandom ? true # enable linux getrandom support
+, es_jitterRng ? true # enable support for the entropy source: jitter rng
+, es_cpu ? true # enable support for the entropy source: cpu-based entropy
+, es_kernel ? true # enable support for the entropy source: kernel-based entropy
+, es_irq ? true # enable support for the entropy source: interrupt-based entropy
+, es_sched ? true # enable support for the entropy source: scheduler-based entropy
+, es_hwrand ? true # enable support for the entropy source: /dev/hwrng
+, hash_sha512 ? false # set the conditioning hash: SHA2-512
+, hash_sha3_512 ? true # set the conditioning hash: SHA3-512
```

Comment:
It is good that we have the comments now but do we truly need this or can just set defaults which people are not expected to change? Are there some flags which you definitely want to have turned on all the time? We could just enable them without an option.

Reply:
To me, the selling point of ESDM is that it is highly configurable to satisfy various (governmental) requirements on random number generators. So it makes sense to expose this configurability to the user.

Reply:
This configuration turns most (meson-build-) options on by default. My reasoning here is, that the only other sensible way would be to turn off all options (except lines 15-16 and 26-27 as we mandate that only one of each is true due to the assertion in lines 30-31) and let have the user configure esdm by themselves.
And considering your questions @SuperSandro2000. I think there is no single option in there, which the user shall not be able to configure.
Having these options and making them configurable for the user makes sense to me.

Reply:
@RaitoBezarius by styling you mean the variables should be `camelCase`? Is there maybe value in keeping them close to their upstream styling or is this so firmly part of the Nix interface that it should use Nix styling?

Reply:
It depends highly on the objective: if you want to read some file with as-is upstream config and load it in the override, then, I'd not be surprised of keeping this style, but add a comment to mention this usecase, if not, conversion to camelCase is highly recommended but not mandatory.

Reply:
I'd say let's go for camelCase then. "highly recommended" is good enough :D. Maybe we can nudge the documentation team that this convention should be integrated into the manual too (I couldn't find anything about this in there).

Reply:
I have now renamed them to use camelCase.